packages/arb-token-bridge-ui/src/util/networks.ts

```diff
@@ -190,3 +219,98 @@ export function getNetworkName(chainId: number) {
       return 'Unknown'
   }
 }
+
+export function getNetworkLogo(chainId: number) {
+  switch (chainId) {
+    // L1 networks
+    case ChainId.Mainnet:
+    case ChainId.Rinkeby:
+    case ChainId.Goerli:
+      return EthereumLogo
+
+    // L2 networks
+    case ChainId.ArbitrumOne:
+    case ChainId.ArbitrumRinkeby:
+    case ChainId.ArbitrumGoerli:
+      return ArbitrumOneLogo
+
+    case ChainId.ArbitrumNova:
+      return ArbitrumNovaLogo
+
+    default:
+      return EthereumLogo
+  }
+}
+
+export type SwitchChainProps = {
+  chainId: number
+  provider: ExtendedWeb3Provider
+  onSuccess?: () => void
+  onError?: (err?: Error) => void
+  onSwitchChainNotSupported?: () => void
```

Comment:
If we update the signature to:

```suggestion
  onSwitchChainNotSupported?: (attemptedChainId: number) => void
```

That would allow us to have a default implementation (instead of noop) that looks like this:

```
function onSwitchChainNotSupported(attemptedChainId: number) {
  const isDeposit = isNetwork(attemptedChainId).isEthereum
  const targetTxName = isDeposit ? 'deposit' : 'withdraw'
  const networkName = getNetworkName(attemptedChainId)

  // TODO: show user a nice dialogue box instead of
  // eslint-disable-next-line no-alert
  alert(
    `Please connect to ${networkName} to ${targetTxName}; make sure your wallet is connected to ${networkName} when you are signing your ${targetTxName} transaction.`
  )
}
```

That cleans up a lot of stuff. We don't have to have the implementation in `TransferPanel`